app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def create
+    user = User.find_by_token(params[:token])
+    if user
+      user.password = params[:password]
+      user.generate_token
```

Comment:
At this point - you definitely want to remove the token rather than generate any other, as a security rule to not leave a possibility for attacker to gain control over the account.